File: backend/src/domain/repository/AuthenticationRepository.ts
import User from "../model/User";
import StateCallback from "../utils/StateCallback";

export default interface AuthenticationRepository extends Repository {

    signIn(userName: string, password: string, callback: StateCallback<boolean>): any;

    signUp(user: User, callback: StateCallback<boolean>): any;

    forgotPassword(email: string): any
}

File: android-app/domain/src/main/java/com/starlight/module/domain/repository/AuthenticationRepository.kt
package com.starlight.module.domain.repository

import com.starlight.module.domain.model.Status
import com.starlight.module.domain.model.User
import com.starlight.module.domain.utils.DataState
import kotlinx.coroutines.flow.Flow

interface AuthenticationRepository {

    fun emailLogin(email: String, password: String): Flow<DataState<Status>>

    fun emailSignUp(user: User, password: String): Flow<DataState<Status>>

    fun refreshToken(refreshToken: String): Flow<DataState<Status>>

    fun logout(): Flow<DataState<Status>>
}